plugins/vpc-shared-eni/network/bridge_linux.go

```diff
@@ -121,36 +124,86 @@ func (nb *BridgeBuilder) FindOrCreateEndpoint(nw *Network, ep *Endpoint) error {
 		return err
 	}
 
+	gatewayIPAddress := nw.GatewayIPAddress
+	var gatewayMACAddress net.HardwareAddr
+
+	// Check whether the endpoint is in the same subnet as the ENI.
+	epSubnetPrefix := vpc.GetSubnetPrefix(ep.IPAddress)
+	eniSubnetPrefix := vpc.GetSubnetPrefix(nw.ENIIPAddress)
+	sameSubnet := (epSubnetPrefix.String() == eniSubnetPrefix.String())
+
+	if nw.BridgeType == config.BridgeTypeL3 || !sameSubnet {
+		// Route ingress traffic for the endpoint to the bridge.
+		route := &netlink.Route{
+			LinkIndex: nw.BridgeIndex,
+			Scope:     netlink.SCOPE_LINK,
+		}
+
+		// If it is in the same subnet, the IP address must have been allocated from a VPC subnet,
+		// so restrict the route to this IP address only. Otherwise, the IP address is from a CIDR
+		// block delegated to this instance, so add a single route for the entire CIDR block.
+		if sameSubnet {
+			dst := *ep.IPAddress
+			_, maskSize := dst.Mask.Size()
+			dst.Mask = net.CIDRMask(maskSize, maskSize)
+			route.Dst = &dst
+		} else {
+			route.Dst = epSubnetPrefix
+		}
+
+		log.Infof("Adding IP route %+v to bridge.", route)
+		err = netlink.RouteAdd(route)
```

Comment:
Can we use `RouteReplace` instead of `RouteAdd`, since it won't fail if the route already exists.